File: src/main/java/dev/obscuria/tooltips/client/tooltip/filter/AlwaysFilter.java
package dev.obscuria.tooltips.client.tooltip.filter;

import com.github.bsideup.jabel.Desugar;
import net.minecraft.item.ItemStack;

@Desugar
public record AlwaysFilter() implements ItemFilter {

    public static final AlwaysFilter INSTANCE = new AlwaysFilter();

    @Override
    public boolean test(ItemStack stack) {
        return true;
    }
}

File: common/src/main/kotlin/dev/obscuria/tooltips/client/filter/AlwaysFilter.kt
package dev.obscuria.tooltips.client.filter

import com.mojang.serialization.Codec
import net.minecraft.world.item.ItemStack

object AlwaysFilter : IItemFilter {

    val CODEC: Codec<AlwaysFilter> = Codec.unit(AlwaysFilter)

    override fun codec(): Codec<AlwaysFilter> = CODEC

    override fun test(stack: ItemStack): Boolean = true
}